tool/tctl/common/user_command.go

```diff
@@ -272,6 +272,12 @@ func (u *UserCommand) Add(ctx context.Context, client auth.ClientI) error {
 	user.SetRoles(u.allowedRoles)
 
 	if err := client.CreateUser(ctx, user); err != nil {
+		if trace.IsAlreadyExists(err) {
+			editUserCmd := fmt.Sprintf("> tctl edit user/%v # edit a YAML representation of the account", u.login)
```

Comment:
This command doesn't work for local users due to an unfortunate RBAC issue. 

Also, should we detect if the existing user is an SSO user? In this case, the proper way to change roles is to update the auth connector, not edit the user.

Reply:
> This command doesn't work for local users due to an unfortunate RBAC issue.

Which issue exactly? It did work on my machine, but I guess there are some edge cases I didn't account for.

I don't have time to add detecting SSO and then maybe a test for that, but we could update the copy to mention that it works only with local users.

Reply:
https://github.com/gravitational/teleport/issues/21598

Reply:
It works when using `tctl` via the local admin identity, but not via an RBAC identity.  This is because we don't have an RBAC representation for the concept of "get user with secrets".

We added the ability to get users with secrets a long time ago because we wanted a quick way to migrate a user and all of their associated secrets to another cluster with a single command, and we did that by letting you bundle user secrets as part of the user resource itself by running `tctl get --with-secrets user/alice@example.com` locally.  Seemed like a good idea at the time, but its created a mess for compatibility.  The `read` verb had an established meaning already and so we needed to scope the permission to local admin instead of `read` like we do with most secrets.  This was the right call, it would be very unexpected/bad if we started letting anyone with `user:read` export password hashes or the like.  The consequence of that is that there is no valid way to represent the ability to get a user "with secrets" remotely.

In retrospect, we should never have permitted bundling user secrets with the actual user resource, and should have instead created a custom "export" command that returned multiple resources.  Resources with a mix of secret and non-secret contents are a never-ending source of annoyance.

Reply:
Alright, I updated the PR to mention `tctl users update` only. It says "To update an existing local user" now.

I feel like `tctl users update` itself should prevent you from updating SSO users or handle it in some other way if there's a valid use case behind updating this kind of resource.

The `# replace roles` comment is there to stress the point that it'll replace the role list and not just add another role. The customer in question probably assumed that `tctl users add foo --roles reviewer` would add the reviewer role to the user.